roles/kafka_controller/tasks/main.yml

```diff
@@ -0,0 +1,460 @@
+    logredactor_policy_refresh_interval_var: "{{logredactor_policy_refresh_interval}}"
+  with_items: "{{ kafka_controller_logredactor_logger_specs_list }}"
+  when: (kafka_controller_custom_log4j|bool) and (logredactor_enabled|bool)
+  tags:
+    - log
+
+- name: Restart kafka broker
+  ansible.builtin.debug:
+    msg: "restarting kafka broker"
+  notify: restart kafka
+  when: (kafka_controller_custom_log4j|bool) and (logredactor_enabled|bool) and (not kafka_controller_skip_restarts|bool)
+  tags:
+    - log
+
+- name: Create Kafka Controller Jolokia Config
+  template:
+    src: kafka_jolokia.properties.j2
+    dest: "{{kafka_controller_jolokia_config}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when: kafka_controller_jolokia_enabled|bool
+  notify: restart kafka
+  diff: "{{ not mask_sensitive_diff|bool }}"
+  tags:
+    - configuration
+
+- name: Create Kafka Controller Jaas Config
+  template:
+    src: kafka_server_jaas.conf.j2
+    dest: "{{kafka_controller.jaas_file}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when:
+    - "'GSSAPI' in kafka_controller_sasl_enabled_mechanisms"
+  notify: restart kafka
+  tags:
+    - configuration
+
+- name: Create Kafka Controller Password File
+  template:
+    src: password.properties.j2
+    dest: "{{kafka_controller.rest_proxy_password_file}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when: kafka_controller_rest_proxy_enabled and (not rbac_enabled or (rbac_enabled and external_mds_enabled)) and kafka_controller_rest_proxy_authentication_type == 'basic'
+  notify: restart kafka
+  diff: "{{ not mask_sensitive_diff|bool }}"
+
+- name: Deploy JMX Exporter Config File
+  template:
+    src: "{{kafka_controller_jmxexporter_config_source_path}}"
+    dest: "{{kafka_controller_jmxexporter_config_path}}"
+    mode: 0640
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+  when: kafka_controller_jmxexporter_enabled|bool
+  tags:
+    - configuration
+
+- name: Create Service Override Directory
+  file:
+    path: "{{kafka_controller.systemd_override | dirname }}"
+    owner: "{{kafka_controller_user}}"
+    group: "{{kafka_controller_group}}"
+    state: directory
+    mode: 0640
+  tags:
+    - systemd
+
+- name: Write Service Overrides
+  template:
+    src: override.conf.j2
+    dest: "{{ kafka_controller.systemd_override }}"
+    mode: 0640
+    owner: root
+    group: root
+  notify: restart kafka
+  diff: "{{ not mask_sensitive_diff|bool }}"
+  tags:
+    - systemd
+
+- name: Create sysctl directory on Debian distributions
+  file:
+    path: /usr/lib/sysctl.d
+    state: directory
+  when: ansible_distribution == "Debian"
+  tags:
+    - sysctl
+    - privileged
+
+- name: Create sysctl directory on Debian distributions
```

Comment:
What is the difference between this task and previous task?

Reply:
I think the difference is of the mode. I don't know if it's required but it is currently used in kafka_broker role

Reply:
I think this might be an outcome of a merge conflict. Let's remove the one w/o permissions. Both here and in the broker.